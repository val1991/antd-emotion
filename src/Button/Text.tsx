/** @jsxImportSource @emotion/react */
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { jsx } from '@emotion/react'
import { Col } from 'antd'
import { PropsWithChildren, ReactElement } from 'react'

export function Text({
  children,
}: PropsWithChildren): ReactElement {


  return (
    <Col
      css={{
        color: 'red',

        '&:active': {
          background: 'green'
        },

        '&:hover': {
          background: 'aqua'
        },
      }}
    >
      {children}
    </Col>
  )
}
