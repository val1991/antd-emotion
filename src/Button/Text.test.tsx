import { render, screen } from '@testing-library/react';
import { Text } from './Text';
import { createSerializer } from '@emotion/jest';
import { StyleProvider } from '@ant-design/cssinjs';

expect.addSnapshotSerializer(createSerializer())

test('Text styles', () => {
  const styleHolder = document.createElement('div');
  const { asFragment } = render(
    <StyleProvider container={styleHolder}>
  <Text>test</Text>
  </StyleProvider>
  );
  expect(asFragment()).toMatchSnapshot()
  
  const text = screen.getByText(/test/i);
  expect(text).toBeInTheDocument();

   expect(text).toHaveStyle({
      color: 'red'
    })
});
