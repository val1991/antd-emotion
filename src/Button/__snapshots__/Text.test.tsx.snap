// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text styles 1`] = `
<DocumentFragment>
  .emotion-0 {
  color: red;
}

.emotion-0:active {
  background: green;
}

.emotion-0:hover {
  background: aqua;
}

<div
    class="ant-col emotion-0 emotion-1"
  >
    test
  </div>
</DocumentFragment>
`;
